Etabs_Adapter/CRUD/Create/Load.cs

```diff
@@ -273,32 +273,85 @@ public void SetLoad(BarPointLoad barPointLoad, bool replace)
 
         /***************************************************/
 
-        public void SetLoad(AreaUniformTemperatureLoad areaTempratureLoad, bool replace)
+        public void SetLoad(AreaUniformTemperatureLoad areaTemperatureLoad, bool replace)
         {
             int ret = 0;
-            string caseName = GetAdapterId<string>(areaTempratureLoad.Loadcase);
-            foreach (IAreaElement area in areaTempratureLoad.Objects.Elements)
+            string caseName = GetAdapterId<string>(areaTemperatureLoad.Loadcase);
+            foreach (IAreaElement area in areaTemperatureLoad.Objects.Elements)
             {
-                double val = areaTempratureLoad.TemperatureChange;
+                double val = areaTemperatureLoad.TemperatureChange;
                 if (val != 0)
                     ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 1, val, "", replace);
             }
         }
 
         /***************************************************/
 
-        public void SetLoad(BarUniformTemperatureLoad barTempratureLoad, bool replace)
+        public void SetLoad(AreaDifferentialTemperatureLoad areaDifferentialTemperatureLoad, bool replace)
         {
             int ret = 0;
-            string caseName = GetAdapterId<string>(barTempratureLoad.Loadcase);
-            foreach (Bar bar in barTempratureLoad.Objects.Elements)
+            string caseName = GetAdapterId<string>(areaDifferentialTemperatureLoad.Loadcase);
+            foreach (IAreaElement area in areaDifferentialTemperatureLoad.Objects.Elements)
             {
-                double val = barTempratureLoad.TemperatureChange;
+                Dictionary<double, double> tempProfile = areaDifferentialTemperatureLoad.TemperatureProfile;
+                double tempMed = tempProfile.Average(kv => kv.Value);
+                double tempDelta= ((tempProfile[0] - tempMed) - (tempProfile[1] - tempMed));
+
+                if (tempMed!=0)
+                { ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 1, tempMed, "", replace);}
+
+                if (tempDelta!=0)
+                { ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 3, tempDelta, "", replace);}
```

Comment:
```suggestion
                if (tempMed!=0)
                  ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 1, tempMed, "", replace);

                if (tempDelta!=0)
                  ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 3, tempDelta, "", replace);
```
If I read this correctly, you are just providing a middle temperature (at the centre of the deck) and then a delta which gets applied positively and negatively?  

If that's the case, I think this only works for linear profiles - if it's a nonlinear profile the `Average` does not suffice as the temperature at the middle. 

I also think it's worth raising an error here to state what is being done. The BHoM representation allows nonlinear temperature profiles but ETABS is limited to linear. So that should be made clear to the user.

Reply:
@peterjamesnugent ,
thanks for your comment.
I've added a warning message for the user in the latest commit.

Reply:
So this just raises a warning no matter what. I think we need to distinguish between `BarDifferentialTemperatureLoad` that either has top/bottom positions, if the differential is not mirrored about the centre and when it has > 2 temperatures. 

1. User pushes `BarDifferentialTemperatureLoad`
a. If it contains 2 `Positions`
i. Check if the average is symmetric about 0 (which I believe ETABS is doing - maybe @Chrisshort92 can confirm?)
ii. Raise error that the temperature is not symmetric about 0 and will not be pushed 
iii. Otherwise push and do not raise warning
b. If it contains > 2 `Positions`
i. Push load 
ii. Raise an warning stating "ETABS does not support nonlinear temperature profiles.

Reply:
Amazing.
Thanks for the comments @peterjamesnugent.
I'll implement them in the code today ;)

Reply:
As discussed offline, happy that the uneven temperature gradients are accounted for using the `tempDelta` to offset the temperature graidnet (`tempMed`). We can remove the warning about uneven temperature distributions, and we can flag a not implemented error for any nonlinear temperature profiles, stating Etabs does not support it.

Reply:
@peterjamesnugent, Error messages updated as per comments and updated also for Area Objects ;)